boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -93,9 +93,17 @@ STRIP = $(CROSSDEV)strip --strip-unneeded
 AR = $(CROSSDEV)ar rcs
 endif
 NM = $(CROSSDEV)nm
+
 OBJCOPY = $(CROSSDEV)objcopy
 OBJDUMP = $(CROSSDEV)objdump
 
+# Native (clang based) compiler doesn't have objcopy ... just ignore it.
+ifeq ($(CONFIG_HOST_MACOS),y)
+  ifeq ($(CONFIG_HOST_ARM64),y)
```

Comment:
we can remove this check?

Reply:
No, as objcopy is not available on MacOS using native compiler. M1/ARM64 support is based on native compiler.

Reply:
I assume macOS/x64 also use native compiler too.

Reply:
No, MacOS/X86 uses the homebrew gcc/binutils. That's also how it's documented in "Getting Started". Removing the guard "ifeq ($(CONFIG_HOST_ARM64),y)" will break the CI build. We had the discussion last week already. Or view was, that it would make sense to get MacOS/ARM and MacOS/X86 simulation on the same baseline --> preferred the native compiler. What I can't forsee is the impact to the community, if the MacOS/X86 simulation is changed from homebrew gcc/binutils to native compiler suite. I mean, if it breaks the workflow of others as they depend on current homebrew gcc/binutils solutions. Any proposal how to progress here and get a good decision?

Reply:
@yamt what your opinion to use native compiler for macOS/x84 by default to simplify the setup? I know user still need gcc to try NuttX ELF feature, but clang is enough for most user case.
If you don't object the change, I would suggest to unify the macOS experience for both x64 and arm64, and simplify the setup process by update CI and documentation.

Reply:
I created separate PR to enable clang based native compiler on MacOS for X86_64/ARM64:
  https://github.com/apache/incubator-nuttx/pull/5452

Reply:
This has been solve in a different PR: https://github.com/apache/incubator-nuttx/pull/5452